### ISSUE TITLE
Add support for conda-lock files

### ISSUE THREAD

**Issue by djhoese** [astropy/ci-helpers#459]
I'm not sure if this fits in with the ci-helpers workflow, but could conda-lock lock files be supported?

https://pypi.org/project/conda-lock/

```
# generate the lockfiles
conda-lock -f environment.yml -p osx-64 -p linux-64

# create an environment from the lockfile
conda create -n my-locked-env --file conda-linux-64.lock
```

To be clear: Just the last command.

**Comment by bsipocz** [comment]
we may already support this unless there is a difference between the lock file and env file specified with `CONDA_ENVIRONMENT`

**Comment by djhoese** [comment]
That's what I though too but when I tried it found out it isn't. The env file has to be passed to conda env create, the lock file is passed to conda create. I think this might require a separate env var. The other question then is should ci-helpers assume that the lock file is a base env and add pytest and all the other dep packages after the env is created? Probably. I guess.

**Comment by djhoese** [comment]
So I did some tests this morning with my own branch where I added a `CONDA_LOCK` variable which is used in place of the current `conda create` and `conda create env` options right now. It still takes forever to solve the environment for other installs: one for conda update, one for pytest and pip, and there are some other ones that I can't quite follow based on the logs I'm seeing on travis. Note my conda dependencies is an empty string so it shouldn't be effecting this. :thinking:
